src/components/Select/Select.tsx

```diff
@@ -2,20 +2,23 @@ import React from 'react'
 import { Typeahead } from 'react-bootstrap-typeahead'
 import 'react-bootstrap-typeahead/css/Typeahead.css'
 
-type SelectOption = { label: string; value: any }
+interface SelectOption<T> {
+  label : string
+  value : T
+}
 
-interface Props {
+interface Props<T> {
   id: string
-  options: SelectOption[]
-  defaultSelected?: SelectOption[]
-  onChange?: (values: string[]) => void
+  options: SelectOption<any>[]
+  defaultSelected?: SelectOption<any>[]
+  onChange?: (values: T[]) => void
   placeholder?: string
   multiple?: boolean
   disabled?: boolean
   isInvalid?: boolean
 }
 
-const Select = (props: Props) => {
+const Select = <T extends {}>(props: Props<T>) => {
```

Comment:
I would prefer to the write this as 

```
function Select<T>(props: Props<T>) {}
```

I could be persuaded otherwise though. Thoughts?

Reply:
Since a majority of components are build using `const` we can keep it that way ?